Sources/TestSupport/SwiftPMProduct.swift

```diff
@@ -84,4 +89,16 @@ public enum SwiftPMProduct {
             throw error
         }
     }
+
+    public static func packagePath(for packageName: String, packageRoot: AbsolutePath) throws -> AbsolutePath {
+        // FIXME: This uses package tool right now, and should be replaced by --get-package-path
+        // whenever we design that.
+        try POSIX.chdir(packageRoot.asString)
+        let packageTool = SwiftPackageTool()
```

Comment:
unfortunately this isn't working out too well when bootstrap script is run from outside of the swiftpm dir/build folder is different because of current assumptions in places like `ToolDefaults` (`static let execBinDir = AbsolutePath(argv0, relativeTo: currentWorkingDirectory).parentDirectory`) So when ToolDefaults is instantiated again by SwiftTool, AbsolutePath asserts out because of invalid path.

I also tried creating and using the Workspace directly here but then some of the manifests fail to load..

There are couple of options:
1. Prioritise designing of get-package-path.
2. Add get-package-path undocumented.
3. Assume the package directory for new and old resolver and return those from here.

What do you think we should do?